control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go

```diff
@@ -128,7 +128,7 @@ func NewParams(hcp *hyperv1.HostedControlPlane, version string, releaseImageProv
 	p.DeploymentConfig.SetDefaultSecurityContext = setDefaultSecurityContext
 	if util.IsPrivateHCP(hcp) {
 		p.APIServerAddress = fmt.Sprintf("api.%s.hypershift.local", hcp.Name)
-		p.APIServerPort = 443
+		p.APIServerPort = util.KASPodPortForPrivateHostedCluster(util.IsRouteKAS(hcp))
```

Comment:
This naming seems misleading. No topology should influence the port the pod. That's an impl detail and we set it via func KASPodPort() already. I assume you are meaning something different here.

Reply:
> I assume you are meaning something different here.

Sorry for the confusion. What I meant is "the port used to communicate with the KAS via the `api.<hc-name>.hypershift.local` endpoint". I'm thinking of `util.APIPortForPrivateHZ(util.IsRouteKAS(hcp))` or `util.APIPortForPrivateDNSHZ(util.IsRouteKAS(hcp))`. What do you think ?

Reply:
Adopted `util.APIPortForLocalZone(util.IsRouteKAS(hcp))`.